internal/generate/clusterserviceversion/clusterserviceversion_updaters.go

```diff
@@ -313,10 +327,16 @@ func validatingToWebhookDescription(webhook admissionregv1.ValidatingWebhook, de
 		seNone := admissionregv1.SideEffectClassNone
 		description.SideEffects = &seNone
 	}
+	if ws != nil && len(ws.Spec.Ports) != 0 {
+		description.ContainerPort = ws.Spec.Ports[0].Port
+	}
 
 	if serviceRef := webhook.ClientConfig.Service; serviceRef != nil {
 		if serviceRef.Port != nil {
-			description.ContainerPort = *serviceRef.Port
+			description.TargetPort = &intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: *serviceRef.Port,
+			}
```

Comment:
I think you need to do something like the following to ensure the description gets the correct port (it can be any port in the array, but container and service port must match)
```suggestion
			var svcPort corev1.ServicePort
			if ws != nil {
				for _, port := range ws.Spec.Ports {
					if *serviceRef.Port == port.Port {
						description.ContainerPort = port.Port
						description.TargetPort = port.TargetPort
						break
					}
				}
			}
```